File: src/main/java/com/refinedmods/refinedstorage/api/network/node/INetworkNodeFactory.java
package com.refinedmods.refinedstorage.api.network.node;

import net.minecraft.core.BlockPos;
import net.minecraft.nbt.CompoundTag;
import net.minecraft.world.level.Level;

import javax.annotation.Nonnull;

/**
 * A factory for reading network nodes from the disk. Used in a {@link INetworkNodeRegistry}.
 */
public interface INetworkNodeFactory {
    /**
     * Creates a network node.
     *
     * @param tag   the tag on disk
     * @param world the world
     * @param pos   the pos
     * @return the network node
     */
    @Nonnull
    INetworkNode create(CompoundTag tag, Level world, BlockPos pos);
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/network/node/INetworkNodeFactory.kt
package com.refinedmods.refinedstorage.api.network.node

import net.minecraft.nbt.CompoundTag
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World


/**
 * A factory for reading network nodes from the disk. Used in a [INetworkNodeRegistry].
 */
interface INetworkNodeFactory {
    /**
     * Creates a network node.
     *
     * @param tag   the tag on disk
     * @param world the world
     * @param pos   the pos
     * @return the network node
     */
    fun create(tag: CompoundTag, world: World, pos: BlockPos): INetworkNode
}